ui/receive_page.go

```diff
@@ -0,0 +1,594 @@
+
+	pageContent := []func(gtx C) D{
+		func(gtx C) D {
+			return pg.pageSections(gtx, func(gtx C) D {
+				return pg.wallAcctWidget.accountSelectSection(gtx, common)
+			})
+		},
+		func(gtx C) D {
+			pg.line.Width, pg.line.Height = gtx.Constraints.Max.X, 1
+			pg.line.Color = common.theme.Color.Background
+			return pg.line.Layout(gtx)
+		},
+		func(gtx C) D {
+			return pg.pageSections(gtx, func(gtx C) D {
+				return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+					layout.Rigid(func(gtx C) D {
+						return pg.generateAddressSection(gtx, common)
+					}),
+					layout.Rigid(func(gtx C) D {
+						return pg.addressSection(gtx, common)
+					}),
+				)
+			})
+		},
+	}
+
+	dims := common.Layout(gtx, func(gtx C) D {
+		return common.UniformPadding(gtx, func(gtx C) D {
+			return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+				layout.Rigid(func(gtx C) D {
+					return layout.Inset{Bottom: values.MarginPadding16}.Layout(gtx, func(gtx C) D {
+						return pg.topNav(gtx, common)
+					})
+				}),
+				layout.Rigid(func(gtx C) D {
+					return common.theme.Card().Layout(gtx, func(gtx C) D {
+						return pg.pageContainer.Layout(gtx, len(pageContent), func(gtx C, i int) D {
+							return pageContent[i](gtx)
+						})
+					})
+				}),
+			)
+		})
+	})
+
+	if pg.wallAcctWidget.isWalletAccountModalOpen {
+		return common.Modal(gtx, dims, pg.wallAcctWidget.walletAccountSection(gtx, common))
+	}
+
+	return dims
+}
+
+func (pg *receivePage) pageSections(gtx layout.Context, body layout.Widget) layout.Dimensions {
+	return pg.theme.Card().Layout(gtx, func(gtx C) D {
+		gtx.Constraints.Min.X = gtx.Constraints.Max.X
+		return layout.UniformInset(values.MarginPadding16).Layout(gtx, body)
+	})
+}
+
+func (pg *receivePage) pageBackDrop(gtx layout.Context) {
+	if pg.isNewAddr {
+		gtx.Constraints.Min.X = gtx.Constraints.Max.X
+		gtx.Constraints.Min.Y = gtx.Constraints.Max.Y
+		m := op.Record(gtx.Ops)
+		pg.backdrop.Layout(gtx)
+		op.Defer(gtx.Ops, m.Stop())
+	}
+}
+
+func (pg *receivePage) topNav(gtx layout.Context, common pageCommon) layout.Dimensions {
+	m := values.MarginPadding20
+	return layout.Flex{}.Layout(gtx,
+		layout.Rigid(func(gtx C) D {
+			return layout.Flex{Axis: layout.Horizontal}.Layout(gtx,
+				layout.Rigid(func(gtx C) D {
+					common.subPageBackButton.Icon = common.icons.contentClear
+					return common.subPageBackButton.Layout(gtx)
+				}),
+				layout.Rigid(func(gtx C) D {
+					txt := common.theme.H6("Receive DCR")
+					txt.Color = common.theme.Color.DeepBlue
+					return layout.Inset{Left: m}.Layout(gtx, txt.Layout)
+				}),
+			)
+		}),
+		layout.Flexed(1, func(gtx C) D {
+			return layout.E.Layout(gtx, func(gtx C) D {
+				return layout.Flex{Axis: layout.Horizontal}.Layout(gtx,
+					layout.Rigid(func(gtx C) D {
+						return common.subPageInfoButton.Layout(gtx)
+					}),
+				)
+			})
+		}),
+	)
+}
+
+func (pg *receivePage) generateAddressSection(gtx layout.Context, common pageCommon) layout.Dimensions {
```

Comment:
This seems like it would be better as "titleSection" as the generate address button is hidden until the button is clicked.